### PR TITLE
Don't emit a space between coords if the next coord is negative

### PR DESCRIPTION
**Pull request by rsheeter**

### CODE REVIEW

**Review by EsoFromTheEast — APPROVED** [review]

File: src/pathstyle.rs [hunk 1]
```diff
@@ -52,6 +52,22 @@ fn coord_string(p: Point) -> String {
     format!("{},{}", round2(p.x), round2(p.y))
 }
 
+struct SvgCoords(String);
```
Comment (cmyr) [inline src/pathstyle.rs:55]: coming into this as a reviewer without any context I would personally benefit from like a line of docs here explaining a bit more about what this is for?

File: src/pathstyle.rs [hunk 2]
```diff
@@ -65,14 +81,14 @@ fn add_command<T, const N: usize>(
     let absolute = coords
         .iter()
         .map(|p| p.write_absolute_coord())
-        .collect::<Vec<_>>()
-        .join(" ");
+        .collect::<SvgCoords>()
```
Comment (cmyr) [inline src/pathstyle.rs:84]: so it looks like we're sort of using the `FromIterator` method as a way of doing a bit of validation or something? It's a slightly funny pattern given that the type we're using is entirely transient, and we're really just trying to convert `Iterator<Item=String>` into `String`? At least given that this is an unusual pattern I would try to do just a bit more to explain it in the docs on `SvgCoords`, and I would personally probably just write a function or something?

**Review by cmyr — APPROVED** [review]
The logic looks good but the overall use of the type is unconventional and a bit confusing, and I think deserves a little bit of explanation... is there some reason to prefer this to just writing a `fn(&[String]) -> String`?

File: src/pathstyle.rs (hunk 2 above)
Comment (rsheeter) [reply src/pathstyle.rs:84]: I wanted to control how an iterator of strings combines to one while keeping the left to right top to bottom reading flow that chaining provides. If you have to wrap a function call around it you lose the readability. I can add comments to that effect.

Comment (cmyr) [reply src/pathstyle.rs:84]: okay sure I think that makes sense, I would just add a comment on `SvgCoords` to that effect.


I know you don't like this particular pattern but if I were writing this something I would personally do is to declare the type and the `FromIterator` impl inside this function, to make it clear that the type only exists to be used here.

Comment (rsheeter) [reply src/pathstyle.rs:88]: Comment added to more clearly indicate it's just to give us something to hang FromIterator on.

File: src/pathstyle.rs (hunk 1 above)
Comment (rsheeter) [reply src/pathstyle.rs:59]: Added docs per IM.

**Pull request merged**